modules/custom/d_demo/d_demo.install

```diff
@@ -1,16 +1,226 @@
+    $storage = \Drupal::entityTypeManager()->getStorage('block_content');
+    $entity = $storage->create($content + ['uuid' => $uuid]);
+    $entity->save();
+  }
+
+  // Place blocks.
+  $i = 0;
+  foreach (d_demo_footer_block_placement() as $region => $blocks) {
+    foreach ($blocks as $block) {
+      Block::create([
+        'id' => "footer_demo_block_$i",
+        'plugin' => $block,
+        'region' => $region,
+        'weight' => 0,
+        'theme' => \Drupal::config('system.theme')->get('default'),
+        'settings' => [
+          'label' => "Demo Block $i",
+          'label_display' => FALSE,
+        ],
+      ])->save();
+
+      $i++;
+    }
+  }
+}
+
+/**
+ * Creates a Menu example block to be placed in the footer.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ * @throws \Drupal\Core\Entity\EntityStorageException
+ */
+function d_demo_create_footer_menu() {
+  \Drupal::entityTypeManager()->getStorage('menu')->create([
+    'id' => 'footer_example_menu',
+    'label' => 'Footer example menu',
+    'description' => 'Footer example menu'
+  ])->save();
+
+  $links = [
+    'Home' => '/',
+    'About' => '/about',
+    'Services' => '/services',
+    'Gallery' => '/gallery',
+    'Contact' => '/contact',
+  ];
+
+  foreach ($links as $label => $location) {
+    MenuLinkContent::create([
+      'title' => $label,
+      'link' => ['uri' => 'internal:' . $location],
+      'menu_name' => 'footer_example_menu',
+      'expanded' => true
+    ])->save();
+  }
+}
+
 /**
  * Implements hook_install().
  */
 function d_demo_install() {
-  if (\Drupal::moduleHandler()->moduleExists('d_social_media')) {
-    $config = \Drupal::configFactory()->getEditable('d_social_media.settings')
-      ->set('link_facebook', 'https://www.facebook.com/dropticacom/')
-      ->set('link_twitter', 'https://twitter.com/droptica')
-      ->set('link_youtube', 'https://www.youtube.com/channel/UCL_XD86NqAJppzw3w7QJqGw')
-      ->set('link_instagram', 'https://www.instagram.com/droptica/')
-      ->set('link_linkedin', 'https://www.linkedin.com/company/2947414')
-      ->save();
+  $config = \Drupal::configFactory()->getEditable('d_social_media.settings')
+    ->set('link_facebook', 'https://www.facebook.com/dropticacom/')
+    ->set('link_twitter', 'https://twitter.com/droptica')
+    ->set('link_youtube', 'https://www.youtube.com/channel/UCL_XD86NqAJppzw3w7QJqGw')
+    ->set('link_instagram', 'https://www.instagram.com/droptica/')
+    ->set('link_linkedin', 'https://www.linkedin.com/company/2947414')
+    ->save();
+
+  d_demo_create_footer_menu();
+  d_demo_create_footer_blocks();
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function d_demo_uninstall() {
+  // Delete blocks.
+  $i = 0;
+  foreach (d_demo_footer_block_placement() as $region => $blocks) {
+    foreach ($blocks as $block) {
```

Comment:
Create a function `d_content_init_create_unplace_block($id)` in `d_content_init` module, perhaps we should hold this ID on the list of blocks and set it as a parameter of `d_content_init_place_block`?